File: frontend/lib/models/response_model.dart
class ResponseModel {
  int? status;
  String? message;
  dynamic data;

  ResponseModel({this.status, this.data});

  ResponseModel.fromJson(Map<String, dynamic> json) {
    status = json['status'];
    message = json['message'];
    data = json['data'];
  }
}

File: backend/src/common/response.model.ts
import { Response } from 'src/types';

export class HttpResponse<T> implements Response<T> {
  status: number;
  message: string;
  data: T;

  constructor(data: T, message?: string, status?: number) {
    this.data = data;
    this.message = message;
    this.status = status;
  }
}
